modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -0,0 +1,753 @@
+
+      out_mesh->stitch_meshes(pattern_mesh,
+                              OUTER_SIDESET_ID,
+                              OUTER_SIDESET_ID,
+                              TOLERANCE,
+                              /*clear_stitched_boundary_ids=*/false);
+
+      // Translate back now that we've stitched so that anyone else that uses this mesh has it at
+      // the origin
+      MeshTools::Modification::translate(pattern_mesh, -(deltax + j * input_pitch), -deltay, 0);
+    }
+  }
+
+  // Check if OUTER_SIDESET_ID is really the external boundary. Correct if needed.
+  auto side_list = out_mesh->get_boundary_info().build_side_list();
+  for (auto & sl : side_list)
+  {
+    if (std::get<2>(sl) == OUTER_SIDESET_ID)
+      if (out_mesh->elem_ptr(std::get<0>(sl))->neighbor_ptr(std::get<1>(sl)) != nullptr)
+        out_mesh->get_boundary_info().remove_side(
+            out_mesh->elem_ptr(std::get<0>(sl)), std::get<1>(sl), std::get<2>(sl));
+  }
+  out_mesh->get_boundary_info().clear_boundary_node_ids();
+
+  out_mesh->get_boundary_info().build_node_list_from_side_list();
+  const auto node_list = out_mesh->get_boundary_info().build_node_list();
+
+  std::vector<Real> bd_x_list;
+  std::vector<Real> bd_y_list;
+  std::vector<std::pair<Real, unsigned int>> node_azi_list;
+  const Point origin_pt = MooseMeshUtils::meshCentroidCalculator(*out_mesh);
+  const Real origin_x = origin_pt(0);
+  const Real origin_y = origin_pt(1);
+
+  MeshTools::Modification::translate(*out_mesh, -origin_x, -origin_y, 0);
+
+  if (_uniform_mesh_on_sides)
+  {
+    MeshTools::Modification::rotate(*out_mesh, -45.0, 0.0, 0.0);
+    const Real azi_tol = 1E-8;
+    for (unsigned int i = 0; i < node_list.size(); ++i)
+    {
+      if (std::get<1>(node_list[i]) == OUTER_SIDESET_ID)
+      {
+        node_azi_list.push_back(
+            std::make_pair(atan2(out_mesh->node_ref(std::get<0>(node_list[i]))(1),
+                                 out_mesh->node_ref(std::get<0>(node_list[i]))(0)) *
+                               180.0 / M_PI,
+                           std::get<0>(node_list[i])));
+        // correct the last node's angle value (180.0) if it becomes a negative value.
+        if (node_azi_list.back().first + 180.0 <= azi_tol)
+          node_azi_list.back().first = 180;
+      }
+    }
+    std::sort(node_azi_list.begin(), node_azi_list.end());
+    const unsigned int side_intervals = node_azi_list.size() / SQUARE_NUM_SIDES;
+    for (unsigned int i = 0; i < SQUARE_NUM_SIDES; i++)
+    {
+      for (unsigned int j = 1; j <= side_intervals; j++)
+      {
+        Real azi_corr_tmp = atan2((Real)j * 2.0 / (Real)side_intervals - 1.0, 1.0);
+        Real x_tmp = _pattern_pitch / 2.0;
+        Real y_tmp = x_tmp * std::tan(azi_corr_tmp);
+        nodeCoordRotate(x_tmp, y_tmp, (Real)i * 90.0 - 135.0);
+        Point p_tmp = Point(x_tmp, y_tmp, 0.0);
+        out_mesh->add_point(p_tmp, node_azi_list[i * side_intervals + j - 1].second);
```

Comment:
why are you adding the point? isnt the node point rotated enough?

Reply:
Just to overwrite the node position because the the node id already exists.